cmd/root.go

```diff
@@ -73,60 +73,143 @@ var (
+			if !cmd.Flags().Changed("log-output") {
+				if envLogOutput, ok := os.LookupEnv("K6_LOG_OUTPUT"); ok {
+					logOutput = envLogOutput
+				}
+			}
+			err := setupLoggers(ctx, logger, fallbackLogger, logFmt, logOutput)
+			if err != nil {
+				return err
 			}
-		}
-		err := setupLoggers(logger, logFmt, logOutput)
-		if err != nil {
-			return err
-		}
 
-		if noColor {
-			// TODO: figure out something else... currently, with the wrappers
-			// below, we're stripping any colors from the output after we've
-			// added them. The problem is that, besides being very inefficient,
-			// this actually also strips other special characters from the
-			// intended output, like the progressbar formatting ones, which
-			// would otherwise be fine (in a TTY).
-			//
-			// It would be much better if we avoid messing with the output and
-			// instead have a parametrized instance of the color library. It
-			// will return colored output if colors are enabled and simply
-			// return the passed input as-is (i.e. be a noop) if colors are
-			// disabled...
-			stdout.Writer = colorable.NewNonColorable(os.Stdout)
-			stderr.Writer = colorable.NewNonColorable(os.Stderr)
-		}
-		stdlog.SetOutput(logger.Writer())
-		logger.Debugf("k6 version: v%s", consts.FullVersion())
-		return nil
-	},
+			if noColor {
+				// TODO: figure out something else... currently, with the wrappers
+				// below, we're stripping any colors from the output after we've
+				// added them. The problem is that, besides being very inefficient,
+				// this actually also strips other special characters from the
+				// intended output, like the progressbar formatting ones, which
+				// would otherwise be fine (in a TTY).
+				//
+				// It would be much better if we avoid messing with the output and
+				// instead have a parametrized instance of the color library. It
+				// will return colored output if colors are enabled and simply
+				// return the passed input as-is (i.e. be a noop) if colors are
+				// disabled...
+				stdout.Writer = colorable.NewNonColorable(os.Stdout)
+				stderr.Writer = colorable.NewNonColorable(os.Stderr)
+			}
+			stdlog.SetOutput(logger.Writer())
+			logger.Debugf("k6 version: v%s", consts.FullVersion())
+			return nil
+		},
+	}
+	return RootCmd
 }
 
 // Execute adds all child commands to the root command sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
-func Execute() {
+func Execute() { //nolint:funlen
+	ctx := context.Background()
+	logger := &logrus.Logger{
+		Out:       os.Stderr,
+		Formatter: new(logrus.TextFormatter),
+		Hooks:     make(logrus.LevelHooks),
+		Level:     logrus.InfoLevel,
+	}
+
+	var fallbackLogger logrus.FieldLogger = &logrus.Logger{
+		Out:       os.Stderr,
+		Formatter: new(logrus.TextFormatter),
+		Hooks:     make(logrus.LevelHooks),
+		Level:     logrus.InfoLevel,
+	}
+	RootCmd := getRootCmd(ctx, logger, fallbackLogger)
```

Comment:
No need to have a capitalized variable here anymore, right?

Reply:
No, they don't. But I wanted the diff to be as easy to read and reason about as possible, so left it  for later. I can rename them in a separate commit if you want, but I prefer the current ones where the differences in the code IMO stays more clear.